CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,287 @@
+```
+git pull upstream master
+```
+
+##### 9. If you have already forked the project, update your copy before working.  
+```  
+$ git remote update  
+$ git checkout <branch-name>  
+$ git rebase upstream/<branch-name>  
+``` 
+## Step:one: Setup the project in your local system :computer:
+
+##### 1.Change working directory to project-
+
+``` 
+cd doc2pen
+```
+
+##### 2. To install all packages 
+
+``` 
+npm install 
+```
+
+##### 3.Local test run
+
+```
+npm start
+```
+##### 4.Now you can now view doc2pen in the browser.
+
+  ``` http://localhost:3000 ```
+
+## step :two:. Identify what you want to contribute :eyes:
+
+- If you want to add or change a major feature, create an issue and discuss with the maintainer what you would like to change.
+- If you want to work on an Existing Issues or an Issue that you created, check if anyone is working on that issue.
+- If no one is working on the issue, just comment and start working on it otherwise wait for the Issue to be assigned to you.
+
+##### 1. Create a new branch :herb:
+
+- Make a branch if you are working on a major features.
+- Use its name to identify the issue your addressing. 
+- DO NOT name it MAIN or MASTER or anything with like 8904thfbn.
+ 
+	###### It will create a new branch with name Branch_Name and switch to that branch
+
+```
+git checkout -b <your_branch_name>
+```
+Our branching strategy is based on [this article](https://nvie.com/posts/a-successful-git-branching-model/) which I suggest you read.
+
++  **master** a history of releases, once merged to from develop and tagged we create a release on the play store & GitHub releases.
+
++  **develop**  the actively worked on next release of the app, what we branch off of while working on new features and what we merge into upon feature completion
+
++ **feature/** or feature/\<username\>/ any branch under this directory is an actively developed feature, feature branches culminate in a PR, are merged and deleted.
+ Typically a feature branch is off of develop and into develop but in rare scenarios if there is an issue in production a branch may be made off master to fix this issue, this type of feature branch must be merged to develop and master before being deleted.
+Branch names should be in the format **\<issue-number\>-kebab-case-title**
+
+All branches should have distinct history and should be visually easy to follow, for this reason only perform merge commits when merging code either by PR or when synchronising.
+
+If you wish to rebase you should be following the [Golden Rule](https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing) and ahere to the advice in the heading [Aside: Rebase as cleanup is awesome in the coding lifecycle](https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase).
+
+## Step:three: Work Remotely 👩‍💻
+- Now you are ready to your work to the remote repository.
+- Perfom your desired changes to the code base.
+- Add all the files/folders needed.  
```

Comment:
Context followed by code or vice versa, that's how technical documentation should be in my opinion